pilot/spec/build_manager_spec.rb

```diff
@@ -1,10 +1,18 @@
+require 'active_support'
+require 'active_support/core_ext'
+
 describe "Build Manager" do
   describe ".truncate_changelog" do
-    it "Truncates Changelog" do
+    it "Truncates Changelog if it exceeds character size" do
       changelog = File.read("./pilot/spec/fixtures/build_manager/changelog_long")
       changelog = Pilot::BuildManager.truncate_changelog(changelog)
       expect(changelog).to eq(File.read("./pilot/spec/fixtures/build_manager/changelog_long_truncated"))
     end
+    it "Truncates Changelog if it exceeds byte size" do
+      changelog = File.read("./pilot/spec/fixtures/build_manager/changelog_bytes_long")
```

Comment:
those might have to be `File.binread` for Windows compat - but not 100% sure

Reply:
Can someone else confirm if this is the case? I do not have a Windows machine to test and all resources I have found speak about reading images, not text. I understand the text has special characters though so it might be needed, any further info?

Reply:
I was just going on the failing AppVeyor test, which is running on a Windows machine:

```
  1) Build Manager .truncate_changelog Truncates Changelog if it exceeds byte size
     Failure/Error: expect(changelog).to eq(File.read("./pilot/spec/fixtures/build_manager/changelog_bytes_long_truncated"))
       expected: "\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6...\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91\xE6\x88\x91..."
            got: "\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u62...\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211\u6211..."
       (compared using ==)
     # ./pilot/spec/build_manager_spec.rb:14:in `block (3 levels) in <top (required)>'
Finished in 2 minutes 31.9 seconds (files took 5.61 seconds to load)
```

So there is definitely something fishy going on there. In the past `binread` was sometimes the solution (although you are right, mostly for binary files) _or_ setting the encoding when reading/writing files (probably to utf-8 in this case) as on Windows ruby's defaults are sometimes different. (So maybe this is an alternative solution that might fix this as well.)

Reply:
Thank you, that is actually also helpful for the solution I am working on, since I have to enforce encoding to UTF-8 so I will have to explore side effects of that enforcing.